### PR TITLE
UI Update

### PR DESCRIPTION
**Pull request by Tanmoy741127**
This issue was originally referenced in  https://github.com/CircuitVerse/CircuitVerse/issues/3381

#### Describe the changes you have made in this PR -
- Updated the UI of Simple Discussion Forum
- Fixed some broken links issue 
- Added an login bar in thread posts page to let users know to login first before commenting. 
- Documentaion updated for customizing themes

### Screenshots of the changes (If any) -
| Page Name | Screenshot |
|-----------|--------------|
| All Threads | image |
| Thread Details | image |
| Post Edit | image |
| New Question | image |
| Login Bar | image | 
| My Questions | image |
| Participating | image |
| Answered | image |
| Unanswered | image |

### CODE REVIEW

**Comment by Tanmoy741127** [comment]
@tachyons 
take a look. 
As per yesterday's discussion, I have done the updates in this repo through thir PR itself.

And also created an PR https://github.com/CircuitVerse/CircuitVerse/pull/3416 in CircuitVerse for necessary updates .

**Comment by vedant-jain03** [comment]
@Tanmoy741127 I have pushed some changes. Check them out and let me know your thoughts on them!

**Comment by Tanmoy741127** [comment]
@vedant-jain03 updated a bit . Now everything is good to go. You can check.

File: app/helpers/simple_discussion/forum_posts_helper.rb [hunk 1]
```diff
@@ -1,9 +1,4 @@
 module SimpleDiscussion::ForumPostsHelper
-  # Override this to use avatars from other places than Gravatar
-  def avatar_tag(email)
```
Comment (tachyons) [inline app/helpers/simple_discussion/forum_posts_helper.rb:3]: Gravatar no longer in use ?

File: app/views/layouts/simple_discussion.html.erb [hunk 2]
```diff
@@ -1,101 +1,68 @@
...
-</div>
-
-<div class="row simple_discussion">
-  <div class="col-md-3 mb-3">
-
-    <div class="card card-body">
-      <%= link_to t('ask_a_question'), simple_discussion.new_forum_thread_path, class: "btn btn-outline-primary btn-block" %>
-      <hr />
+<div>
+  <h1 class="community-heading">Community</h1>
```
Comment (tachyons) [inline app/views/layouts/simple_discussion.html.erb:2]: Make this text i18n compatiable

File: app/views/simple_discussion/forum_threads/_forum_thread.html.erb [hunk 3]
```diff
@@ -1,37 +1,43 @@
...
-        <h4>
           <% if forum_thread.solved? %>
-            <span class="text-success"><%= icon "fas", "check-circle" %></span>
-          <% end %>
-
-          <%= link_to simple_discussion.forum_thread_path(forum_thread) do %>
-            <%= icon "fas", "thumb-tack", class: "text-muted" if forum_thread.pinned? %> <%= forum_thread.title %>
+            <div class="solved-tag">
+              <%= icon "far", "check-circle" %>
+              Solved
```
Comment (tachyons) [inline app/views/simple_discussion/forum_threads/_forum_thread.html.erb:20]: Make this i18n compatiable

File: app/views/simple_discussion/forum_threads/_login_bar.erb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<div class="d-flex flex-row justify-content-center align-items-center mt-4">
+    <h5 class="p-0 m-0"><%= t('want_to_join_the_discussion') %> ?</h5>
+    <a href="/users/sign_up" class="btn forum-primary-btn mx-2"><%= t('create_account') %></a>
```
Comment (tachyons) [inline app/views/simple_discussion/forum_threads/_login_bar.erb:3]: Use devise route helpers

File: app/helpers/simple_discussion/forum_posts_helper.rb (hunk 1 above)
Comment (Tanmoy741127) [reply app/helpers/simple_discussion/forum_posts_helper.rb:3]: Gravatar on use, this specific avatar_tag is not required. Using the `gravatar_url_for` function defined inthatt same file

**Comment by tachyons** [comment]
@Tanmoy741127 There are a few more places where strings are hardcoded, could you convert them to i18n format ?

**Comment by vedant-jain03** [comment]
Screenshot from 2022-12-25 20-27-21

The `undo` button needs to be refactored!

**Comment by Tanmoy741127** [comment]
@tachyons sure.  I will check and convert them to i18n compatible.

**Comment by Tanmoy741127** [comment]
> Screenshot from 2022-12-25 20-27-21
> 
> The `undo` button needs to be refactored!

What changes you prefer ?

**Comment by Tanmoy741127** [comment]
> @tachyons sure. I will check and convert them to i18n compatible.

Changes the other strings to i18n compaitable one in  last commit.

**Comment by Tanmoy741127** [comment]
> Screenshot from 2022-12-25 20-27-21
> 
> The `undo` button needs to be refactored!

@vedant-jain03 
image
image

Can we use that one or its outlined version ?

**Comment by vedant-jain03** [comment]
Screenshot from 2022-12-25 22-02-36

We have used this button in CircuitVerse groups!

**Comment by Tanmoy741127** [comment]
> Screenshot from 2022-12-25 22-02-36
> 
> We have used this button in CircuitVerse groups!

Ohh okk

File: app/views/simple_discussion/user_mailer/new_post.html.erb [hunk 5]
```diff
@@ -2,11 +2,11 @@
   <%= gravatar_image_tag @forum_post.user.email, style: "float: left" %>
 
   <div style="margin-left:60px">
-    <p><strong><%= @forum_post.user.name %></strong> <small>commented:</small></p>
+    <p><strong><%= @forum_post.user.name %></strong> <small><%= t("commented:") %></small></p>
```
Comment (vedant-jain03) [inline app/views/simple_discussion/user_mailer/new_post.html.erb:5]:
```suggestion
    <p><strong><%= @forum_post.user.name %></strong> <small><%= t("commented") %></small></p>
```

The key should be defined in the `en.yml`.
Keep this for other strings as well!

**Review by vedant-jain03 — CHANGES_REQUESTED** [review]

File: app/views/simple_discussion/user_mailer/new_post.html.erb (hunk 5 above)
Comment (Tanmoy741127) [reply app/views/simple_discussion/user_mailer/new_post.html.erb:5]: ok!

Comment (Tanmoy741127) [reply app/views/simple_discussion/user_mailer/new_post.html.erb]: @vedant-jain03 updated in latest commit

**Comment by Tanmoy741127** [comment]
forum-first-page (2)

@nitin10s has suggested to do the changes (moving the category tags , profile pictures position and removal of unsolved/open tag)
@vedant-jain03 @tachyons  can you see the design attached and confirm any other changes required or not ?

**Comment by vedant-jain03** [comment]
> forum-first-page (2)
> 
> @nitin10s has suggested to do the changes (moving the category tags , profile pictures position and removal of unsolved/open tag) @vedant-jain03 @tachyons can you see the design attached and confirm any other changes required or not ?

LGTM, could you commit the changes?

**Comment by tachyons** [comment]
@Tanmoy741127 @vedant-jain03 Merged this PR to avoid PR growing too big to review. Please make follow-up PRs for rest of the changes

**Comment by vedant-jain03** [comment]
@Tanmoy741127 Great job, thanks 💯

**Comment by Tanmoy741127** [comment]
> > forum-first-page (2)
> > @nitin10s has suggested to do the changes (moving the category tags , profile pictures position and removal of unsolved/open tag) @vedant-jain03 @tachyons can you see the design attached and confirm any other changes required or not ?
> 
> LGTM, could you commit the changes?



> @Tanmoy741127 Great job, thanks 100

Thanks ! Can you open the issues tab for this repo, it can be easy to discuss.

**Pull request merged**